tritium-lib/src/main/java/com/palantir/tritium/proxy/ByteBuddyInstrumentation.java

```diff
@@ -202,6 +202,10 @@ private ByteBuddyInstrumentation() {
         checkState(interfaceClass.equals(discoveredInterfaces[0]), "Expected the provided interface first");
         for (int i = 1; i < discoveredInterfaces.length; i++) {
             Class<?> additionalInterface = discoveredInterfaces[i];
+            if (assignableFromAny(additionalInterface, discoveredInterfaces)) {
```

Comment:
In a future change it might be helpful to sort the additional interfaces to avoid re-defining effectively identical classes. The first (requested) interface class order matters, jax-rs annotations are scraped that way, for example, but beyond that there isn't much difference.